osu.Game/Overlays/Dialog/PopupDialog.cs

```diff
@@ -205,8 +207,17 @@ protected PopupDialog()
                     },
                 },
             };
+
+            // It's important we start in a visible state so our state fires on hide, even before load.
+            // This is used by the DialogOverlay to know when the dialog was dismissed.
+            Show();
```

Comment:
I'm not sure I get what's happening here. From what I can gather this and the `StartHidden` specification are two parts of a particular setup, but I'm not sure what this setup is supposed to achieve. If I understand correctly:

* `VisibilityContainer`s already start hidden by default
* `StartHidden` makes super absolutely sure that they start hidden by default and don't play any animations to boot by finishing transforms, but also doesn't change state

so this is basically a workaround to have `State` be `Visible` in ctor, then have `VisibilityContainer.LoadAsyncComplete()` play the pop-out instantly but keep the `Visible` state, and then finally `LoadComplete()` run `UpdateState()` and therefore also run the pop-in animation? And with this setup any changes in the ctor-to-`LoadComplete()` time interval can be detected while they would be previously lost?

If the above is right, then even though the end goal I can agree with, I'm not entirely sure how to feel about the path there. Feels like a rube goldberg type solution if I'm honest.

Reply:
`StartHidden` is strictly animation related. It makes the "pop in" animation play on first display, rather than immediately showing the shown state.

But yes, in practice it's doing exactly what you mention.

Reply:
I think most of my confusion is in that visibility container members are just confusingly named - enabling `StartHidden` feels like it should also mutate `State`, but it doesn't.

That's not in scope here, though, so I guess I could personally live with it at the end of the day - would maybe leave a comment at that `StartHidden` override that it's related to the show call, since it might not be as obvious N months down the line...

Reply:
Added a comment.